pig/src/org/partiql/pig/domain/Utils.kt

```diff
@@ -0,0 +1,67 @@
+package org.partiql.pig.domain
+
+import com.amazon.ionelement.api.IonElement
+import com.amazon.ionelement.api.ionSexpOf
+import com.amazon.ionelement.api.ionSymbol
+import com.amazon.ionelement.api.withAnnotations
+import org.partiql.pig.domain.model.DataType
+import org.partiql.pig.domain.model.NamedElement
+import org.partiql.pig.domain.model.TupleType
+import org.partiql.pig.domain.model.TypeDomain
+
+/**
+ *  Filter a [List]<[TypeDomain]> to only contain [TypeDomain]s that are present in [domains]
+ */
+fun List<TypeDomain>.filterDomains(domains: Set<String>?) =
```

Comment:
This one is a unrelated to the others--can you add this as a non-extension function on `TypeUniverse` (not sure that making an extension function adds much) or at least move it to `src/org.partiql.pig/domain/Utils.kt`?

Reply:
It was used in a couple places before, I just inlined it :D